### PR TITLE
Add Concourse Jobs

### PR DESCRIPTION
**Pull request by liamraemclauchlan**
* Move back to crown concourse
* Add jobs for development, qa and int

### CODE REVIEW

File: aviator.yml [hunk 1]
```diff
@@ -7,8 +7,9 @@ spruce:
...
   - with_in: ci/jobs/
   to: aviator_pipeline.yml
+
 fly:
   name: dataworks-analytical-service-infra
-  target: aws-concourse
+  target: concourse
   expose: false
-  config: aviator_pipeline.yml
+  config: aviator_pipeline.yml
```
Comment (dan-hill2802) [inline aviator.yml]: new line needed

File: ci/jobs/development.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  plan:
+  - get: dataworks-analytical-service-infra
+    trigger: true
+  - .: (( inject meta.plan.terraform-bootstrap ))
+    params:
+      DEPLOY_PATH: infra
+  - .: (( inject meta.plan.terraform-apply ))
+    config:
+      run:
+        dir: 'dataworks-analytical-service-infra/terraform/deploy/infra'
```
Comment (dan-hill2802) [inline ci/jobs/development.yml]: this should be in the meta file, it doesn't vary per job

File: ci/jobs/development.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  - .: (( inject meta.plan.terraform-plan ))
+    config:
+      run:
+        dir: 'dataworks-analytical-service-infra/terraform/deploy/infra'
```
Comment (dan-hill2802) [inline ci/jobs/development.yml]: this should be in the meta file, it doesn't vary per job

File: ci/meta.yml [hunk 4]
```diff
@@ -24,6 +24,134 @@ meta:
         outputs:
           - name: terraform-config
       params:
-        AWS_ACCESS_KEY_ID: ((dataworks-secrets.ci_aws_access_key_id))
-        AWS_SECRET_ACCESS_KEY: ((dataworks-secrets.ci_aws_secret_access_key))
+        AWS_ACCESS_KEY_ID: ((ci.aws_access_key_id))
+        AWS_SECRET_ACCESS_KEY: ((ci.aws_secret_access_key))
```
Comment (dan-hill2802) [inline ci/meta.yml:29]: do you need this, can you use the instance profile

File: ci/resources.yml [hunk 5]
```diff
@@ -2,6 +2,17 @@ resources:
   - name: dataworks-analytical-service-infra-pr
     type: pull-request
     source:
-      access_token: ((dataworks-secrets.concourse_github_pat))
-      repository: dwp/dataworks-analytical-service-infra
-    check_every: 1m
+      access_token: ((github_com.token))
+      repo: dwp/dataworks-analytical-service-infra
+      uri: https://github.com/dwp/dataworks-analytical-service-infra.git
+    check_every: 15m
```
Comment (dan-hill2802) [inline ci/resources.yml:8]: you won't need the checking, you can use a webhook now

**Review by dan-hill2802 — CHANGES_REQUESTED** [review]

File: aviator.yml (hunk 1 above)
Comment (liamraemclauchlan) [reply aviator.yml]: New line added

File: ci/jobs/development.yml (hunk 2 above)
Comment (liamraemclauchlan) [reply ci/jobs/development.yml]: done - moved to meta.yml

File: ci/jobs/development.yml (hunk 3 above)
Comment (liamraemclauchlan) [reply ci/jobs/development.yml]: done - moved to meta.yml

File: ci/resources.yml (hunk 5 above)
Comment (liamraemclauchlan) [reply ci/resources.yml:8]: Our other repos still use the check - keeping it consistent for now, and will raise a ticket to use webhooks in the future (and retroactively)

File: ci/meta.yml (hunk 4 above)
Comment (liamraemclauchlan) [reply ci/meta.yml:29]: this is crown concourse.

File: ci/jobs/integration.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+jobs:
+- name: integration
+  plan:
+  - get: dataworks-analytical-service-infra
+    trigger: true
+    passed:
+      - qa
+  - .: (( inject meta.plan.terraform-bootstrap ))
+    params:
+      DEPLOY_PATH: infra
```
Comment (dan-hill2802) [inline ci/jobs/integration.yml]: this never varies, so set it in `meta.yml`

File: ci/meta.yml [hunk 7]
```diff
@@ -24,6 +24,136 @@ meta:
...
+          dir: 'dataworks-analytical-service-infra/terraform/deploy/infra'
+          path: sh
+          args:
+            - -exc
+            - |
+              cp ../../../../terraform-config/terraform.tf .
+              cp ../../../../terraform-config/terraform.tfvars .
+              terraform init
+              terraform workspace show
+              terraform plan -var assume_role=ci -out terraform.plan
```
Comment (dan-hill2802) [inline ci/meta.yml]:
```suggestion
              terraform plan -out terraform.plan
```
and use the `sed` command to remove assume role from bootstrap (same as the other pipelines)

File: ci/meta.yml [hunk 8]
```diff
@@ -24,6 +24,136 @@ meta:
...
+          dir: 'dataworks-analytical-service-infra/terraform/deploy/infra'
+          path: sh
+          args:
+            - -exc
+            - |
+              cp ../../../../terraform-config/terraform.tf .
+              cp ../../../../terraform-config/terraform.tfvars .
+              terraform init
+              terraform workspace show
+              terraform plan -var assume_role=ci -detailed-exitcode
```
Comment (dan-hill2802) [inline ci/meta.yml]:
```suggestion
              terraform plan -detailed-exitcode
```

File: ci/meta.yml [hunk 9]
```diff
@@ -24,6 +24,136 @@ meta:
...
+        image_resource:
+          type: docker-image
+          source:
+            repository: ((terraform.repository))
+            version: ((terraform12.version))
+            tag: ((terraform12.version))
+        params:
+          AWS_ACCESS_KEY_ID: ((ci.aws_access_key_id))
+          AWS_SECRET_ACCESS_KEY: ((ci.aws_secret_access_key))
+          TF_INPUT: false
```
Comment (dan-hill2802) [inline ci/meta.yml:77]:
```suggestion
          TF_INPUT: false
          TF_VAR_assume_role: ci
```

File: ci/resources.yml (hunk 5 above)
Comment (dan-hill2802) [reply ci/resources.yml:8]: now that it's in Crown Concourse, you will have to check

**Review by dan-hill2802 — CHANGES_REQUESTED** [review]

File: ci/meta.yml (hunk 7 above)
Comment (liamraemclauchlan) [reply ci/meta.yml]: Done 👍

File: ci/meta.yml (hunk 8 above)
Comment (liamraemclauchlan) [reply ci/meta.yml]: ditto

File: ci/meta.yml (hunk 9 above)
Comment (liamraemclauchlan) [reply ci/meta.yml:77]: Cheers 👍

File: ci/resources.yml [hunk 10]
```diff
@@ -2,6 +2,17 @@ resources:
   - name: dataworks-analytical-service-infra-pr
     type: pull-request
     source:
-      access_token: ((dataworks-secrets.concourse_github_pat))
-      repository: dwp/dataworks-analytical-service-infra
-    check_every: 1m
+      access_token: ((github_com.token))
+      repo: dwp/dataworks-analytical-service-infra
+      uri: https://github.com/dwp/dataworks-analytical-service-infra.git
```
Comment (dan-hill2802) [inline ci/resources.yml:7]: this isn't needed for this resource

File: ci/resources.yml [hunk 11]
```diff
@@ -2,6 +2,17 @@ resources:
   - name: dataworks-analytical-service-infra-pr
     type: pull-request
     source:
-      access_token: ((dataworks-secrets.concourse_github_pat))
-      repository: dwp/dataworks-analytical-service-infra
-    check_every: 1m
+      access_token: ((github_com.token))
+      repo: dwp/dataworks-analytical-service-infra
```
Comment (dan-hill2802) [inline ci/resources.yml:6]:
```suggestion
      repository: dwp/dataworks-analytical-service-infra
```

**Review by dan-hill2802 — CHANGES_REQUESTED** [review]

File: ci/resources.yml (hunk 10 above)
Comment (liamraemclauchlan) [reply ci/resources.yml:7]: changed resource to mirror other repos

File: ci/resources.yml (hunk 11 above)
Comment (liamraemclauchlan) [reply ci/resources.yml:6]: :+1: fixed

File: ci/resource_types.yml [hunk 12]
```diff
@@ -1,6 +1,5 @@
 resource_types:
-- name: pull-request
-  type: docker-image
-  source:
-    repository: teliaoss/github-pr-resource
-    tag: latest
+  - name: pull-request
+    type: docker-image
+    source:
+      repository: jtarchie/pr
```
Comment (dan-hill2802) [inline ci/resource_types.yml]: no new line, REJECTED!

Comment (liamraemclauchlan) [reply ci/resource_types.yml]: 😁  fixing

File: ci/resources.yml [hunk 13]
```diff
@@ -2,6 +2,17 @@ resources:
   - name: dataworks-analytical-service-infra-pr
     type: pull-request
     source:
-      access_token: ((dataworks-secrets.concourse_github_pat))
+      access_token: ((github_com.token))
       repository: dwp/dataworks-analytical-service-infra
```
Comment (dan-hill2802) [inline ci/resources.yml]: this needs to be `repo` for the `jtarchie/pr` resource

**Review by dan-hill2802 — CHANGES_REQUESTED** [review]

File: ci/resources.yml (hunk 13 above)
Comment (liamraemclauchlan) [reply ci/resources.yml]: 🤦

**Review by dan-hill2802 — APPROVED** [review]

**Pull request merged**